crates/astria-sequencer/src/api_state_ext.rs

```diff
@@ -375,3 +375,413 @@ pub(crate) trait StateWriteExt: StateWrite {
+        Protobuf,
+    };
+    use cnidarium::StateDelta;
+    use sha2::{
+        Digest as _,
+        Sha256,
+    };
+    use tendermint::{
+        account,
+        block::{
+            header::Version,
+            Header,
+        },
+        AppHash,
+        Hash,
+        Time,
+    };
+    use tendermint_proto::v0_37::types::{
+        BlockId,
+        PartSetHeader,
+    };
+
+    use super::*;
+    use crate::proposal::commitment::generate_rollup_datas_commitment;
+
+    struct ValueGenerator {
+        value: u8,
+    }
+
+    // used to ensure that unique data is being written for all sequencer block components
+    impl ValueGenerator {
+        fn new(start: u8) -> Self {
+            ValueGenerator {
+                value: start,
+            }
+        }
+
+        fn generate(&mut self) -> u8 {
+            let prev = self.value;
+            // Check if value is at its max and prevent overflow
+            self.value = self.value.saturating_add(1);
+            prev
+        }
+    }
+
+    // creates new sequencer block, optionally shifting all values except the height by 1
+    fn make_test_sequencer_block(gen: &mut ValueGenerator, shift_values: bool) -> SequencerBlock {
+        // create height
+        let height = u32::from(gen.generate());
+
+        // shift values (used for overwriting height test)
+        if shift_values {
+            gen.generate();
+        }
+
+        // create last block id
+        let last_block_id = BlockId {
+            hash: [gen.generate(); 32].to_vec(),
+            part_set_header: Some(PartSetHeader {
+                total: u32::from(gen.generate()),
+                hash: [gen.generate(); 32].to_vec(),
+            }),
+        };
+
+        // create cometbft header
+        let mut header = Header {
+            app_hash: AppHash::try_from([gen.generate(); 32].to_vec()).unwrap(),
+            chain_id: gen.generate().to_string().try_into().unwrap(),
+            consensus_hash: Hash::try_from([gen.generate(); 32].to_vec()).unwrap(),
+            data_hash: Some(Hash::try_from([gen.generate(); 32].to_vec()).unwrap()),
+            evidence_hash: Some(Hash::try_from([gen.generate(); 32].to_vec()).unwrap()),
+            height: height.into(),
+            last_block_id: Some(tendermint::block::Id::try_from(last_block_id).unwrap()),
+            last_commit_hash: Some(Hash::try_from([gen.generate(); 32].to_vec()).unwrap()),
+            last_results_hash: Some(Hash::try_from([gen.generate(); 32].to_vec()).unwrap()),
+            next_validators_hash: Hash::try_from([gen.generate(); 32].to_vec()).unwrap(),
+            proposer_address: account::Id::try_from([gen.generate(); 20].to_vec()).unwrap(),
+            time: Time::now(),
+            validators_hash: Hash::try_from([gen.generate(); 32].to_vec()).unwrap(),
+            version: Version {
+                app: u64::from(gen.generate()),
+                block: u64::from(gen.generate()),
+            },
+        };
```

Comment:
if the generator is used for creating essentially random numbers, maybe use rand instead? https://docs.rs/rand/latest/rand/ then you can do:
```
app_hash: rng.gen(),
...
```

for example

Reply:
Oh nice, I'll replace with rng(). I was trying to avoid doing all zeros/default values because I wanted the tests to ensure that unique data was being written and retrieved from the correct places.